pkg/controllers/awsloadbalancercontroller/rbac_rolebinding.go

```diff
@@ -0,0 +1,137 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+func (r *AWSLoadBalancerControllerReconciler) ensureRoleBinding(ctx context.Context, sa *corev1.ServiceAccount, controller *albo.AWSLoadBalancerController) error {
+	reqLogger := log.FromContext(ctx)
+
+	desired := desiredRoleBinding(ctx, r.Namespace, sa)
+	reqLogger.Info("ensuring rolebindings", "rolebindings", desired.Name)
+
+	if err := controllerutil.SetControllerReference(controller, desired, r.Scheme); err != nil {
+		return fmt.Errorf("failed to set the controller reference for rolebindings %s : %w", desired.Name, err)
+	}
+
+	exist, current, err := r.currentRoleBinding(ctx, desired)
+	if err != nil {
+		return fmt.Errorf("failed to fetch current rolebindings: %w", err)
+	}
+
+	if !exist {
+		if err := r.createRoleBinding(ctx, desired); err != nil {
+			return err
+		}
+
+		_, current, err = r.currentRoleBinding(ctx, desired)
+		if err != nil {
+			return fmt.Errorf("failed to fetch created rolebindings: %w", err)
+		}
+		reqLogger.Info("created rolebindings", "rolebindings", desired.Name)
+	}
+
+	if err := r.updateRoleBinding(ctx, current, desired); err != nil {
+		return fmt.Errorf("failed to update rolebindings: %w", err)
+	}
+
+	return nil
+}
+
+// updateRoleBindings updates the current rolebindings and returns a flag to denote if the update was done.
+func (r *AWSLoadBalancerControllerReconciler) updateRoleBinding(ctx context.Context, current *rbacv1.RoleBinding, desired *rbacv1.RoleBinding) error {
+	changed, err := isRoleBindingChanged(current, desired)
+	if err != nil {
+		return err
+	}
+
+	if !changed {
+		return nil
+	}
+	updated := current.DeepCopy()
+	updated.Name = desired.Name
+	updated.Namespace = desired.Namespace
```

Comment:
Setting name and namespace is not required because it's not possible. Also update with a new name and namespace will fail.

Reply:
Addressed.